runtime/include/chpl-mem-desc.h

```diff
@@ -97,7 +97,10 @@ extern "C" {
   m(MLI_DATA,             "multilocale interop data",                 true ), \
   m(GPU_DATA,             "user data allocation on gpu",              true ), \
   m(GPU_KERNEL_ARG,       "kernel data allocation on gpu",            true ), \
-  m(NUM,                  "*** this must be the last entry ***",      true )
+  m(GPU_KERNEL_PARAM,      "pointer to a kernel arg",                 true ), \
```

Comment:
Maybe to prevent confusion `GPU_KERNEL_ARG` can be renamed to `GPU_OFFLOADED_KERNEL_PARAM`? I had to look at the code to understand the difference between `GPU_KERNEL_PARAM` vs `GPU_KERNEL_ARG`.